### ISSUE TITLE
Versioning

### ISSUE THREAD

**Issue by Newobc** [G-Bro/OpenDM-Server#21]
Do we want to implement a versioning system. Semantic or otherwise, I'm happy to do the old `hotifx-oh-shit-ive-done-it-again`. But, this could be an opportunity to try something new?

**Comment by G-Bro** [comment]
Could do. How do you propose we handle the versioning? I think hotfix and feat branches are still valid. Hotfix increases the patch number, feat increases the minor or major number.

But what happens when two people want to introduce a feature? Merge them together and bump up one minor version or bump up 2 minor versions?

**Comment by Newobc** [comment]
> But what happens when two people want to introduce a feature? Merge them together and bump up one minor version or bump up 2 minor versions?

I would merge them together whenever possible, I don't see a problem with that. Who will do the bumping?

> Could do. How do you propose we handle the versioning? I think hotfix and feat branches are still valid. Hotfix increases the patch number, feat increases the minor or major number.

I was thinking that we do something like this:

`{github_user_name}:hotfix|feature:change-description`

example:

`newobc:feature:cantrip-logic-optimisations`

What do you think?

**Comment by G-Bro** [comment]
Yeah that branch format is cool. As for bumping I'm not sure. I guess we don't merge directly into master, we merge into 'v0.2.0' and then merge that into master when we have achieved everything we planned for that update. Any fixes and minor tweaks get merged into v0.1.(n + 1)

I know that in the npm side you can bump your version number up in package.json, but I don't know if Laravel has something similar we can use.

**Comment by DanEvans485** [comment]
I disagree with user name in the branch, Don't see why we'd need that.

But the current version is 1.1.something I guess. Feature increases minor, Patch / hotfix increases Patch

**Comment by G-Bro** [comment]
We should use GitHub's milestone feature to manage the minor versions. So for example once the 'Monster' milestone is complete we are on v0.1.0

**Comment by G-Bro** [comment]
I'm not particularly bothered about user names in the branch. I guess it makes sense not to have them because branches can be collaborative but I really don't mind. I do like having colons instead of dashes

so we could just go with [type]:[description]

where type is one of feature, hotfix, refactor, chore?

**Comment by Newobc** [comment]
Only suggested because its a common pattern with open source projects.

I dont think you need to have a strict list, in my mind you could also have `enhancement`, `optmisation`... that kind of thing.

**Comment by G-Bro** [comment]
I think we can start implementing versioning once we complete the v0.1 milestone

**Comment by DanEvans485** [comment]
Agreed. Atm every things a bit fixy patchy featcha fixy

**Comment by G-Bro** [comment]
I also think that open-dm/web and open-dm/api should be version synced, so we don't launch v0.2 until we have all the frontend stuff sorted for v0.2 too. That way, if we miss anything then we should be able to patch it in pre-launch

**Comment by DanEvans485** [comment]
But then you have to increase the version of the api when you patch the frontend.

**Comment by G-Bro** [comment]
I meant to say the major and minor versions should be synced. Patch numbers can be different
